src/v/cloud_storage/remote_segment.cc

```diff
@@ -323,8 +389,10 @@ ss::future<uint64_t> remote_segment::put_segment_in_cache_and_create_index(
     }
     if (index_prepared) {
         auto index_stream = make_iobuf_input_stream(tmpidx.to_iobuf());
-        co_await _cache.put(generate_remote_index_path(_path), index_stream);
+        co_await _cache.put(generate_index_path(_path), index_stream);
         _index = std::move(tmpidx);
+        _coarse_index.emplace(_index->build_coarse_index(_chunk_size));
+        co_await _chunks_api->start();
```

Comment:
This function is called from an `is_legacy_mode_engaged` branch. Does it make sense to build the coarse index and start `_chunks_api` here?

Reply:
yeah, we do not need these actions here, fixed.